api/metrics/src/main/java/io/opentelemetry/api/metrics/GlobalMeterProvider.java

```diff
@@ -5,81 +5,30 @@
 
 package io.opentelemetry.api.metrics;
 
+import io.opentelemetry.api.metrics.internal.NoopMeterProvider;
 import java.util.concurrent.atomic.AtomicReference;
 
-/**
- * IMPORTANT: This is a temporary class, and solution for the metrics package until it will be
- * marked as stable.
- */
-public final class GlobalMeterProvider {
-  private static final Object mutex = new Object();
-  private static final AtomicReference<MeterProvider> globalMeterProvider = new AtomicReference<>();
+/** This class is a temporary solution until metrics SDK is marked stable. */
+public class GlobalMeterProvider {
+  private static final AtomicReference<MeterProvider> globalMeterProvider =
```

Comment:
Can this just be a normal `volatile` since we only set and get?

Reply:
The only difference between `volatile` and what I have is the *reader* pays the write-cache-flush+reload cost, while the writer can go on their merry way without a full sync.   Probably WAY to high on the micro-optimizatioon, just standard practice for concurrent programming on the JVM for when I used to do it often.

Reply:
Yeah - since there isn't really performance impact (if there is it's on the get side, not set) we should err on simpler code. In particular, we make it harder for future readers who will need to lookup javadoc to understand `lazySet`, a very uncommonly used method compared to `volatile` which is well understood. This code is going to be deleted anyways, which is probably more the reason to just keep things as simple as possible right now.

Reply:
Sure, but so we're clear, the performance improvement is on the *set* side, not the get.  The get will have a memory barrier that flushes cache, `lazySet` vs. `set` (or volatile write) is cache-flush vs. not cache-flush, so this only impacts speed of writing the global, allowing (e.g.) the agent to consider processing its startup rather than flushing right away.

`lazySet` is pretty common thing for high performance concurrency.  e.g. look at JCTools: https://github.com/JCTools/JCTools/search?q=lazySet
I do *not* think we should shy away from it.  Java's `volatile` feature is pretty poor for modern architecture.    I'm ok migrating this specific code to `volatile`, but want to make sure we aren't preferring `volatile`, generally.

Reply:
This is updated, but leaving unresolved because I'm concerned about the`lazySet` here.  I think usage of `lazySet` is critical to most efficient concurrent solutions in Java and I don't want us shying away from it when it's warranted.

Reply:
Yeah - any app sets a global provider (or OpenTelemetry) once at the beginning of the app initialization so the performance of set doesn't matter at all here. We might also trick something into thinking it does matter if we use `lazySet` here.

Where it does have an effect we would use it and probably add comments on the motivation - in many cases we probably would just end up using it via some other library like jctools though.

Reply:
Sounds good.